staging/src/k8s.io/controller-manager/pkg/healthz/handler_test.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package healthz
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+type badChecker struct{}
+
+func (b *badChecker) Name() string {
+	return "bad"
+}
+
+func (b *badChecker) Check(_ *http.Request) error {
+	return fmt.Errorf("bad")
+}
+
+func TestMutableHealthzHandler(t *testing.T) {
```

Comment:
This is a good test. Also directly validate that two checks can be in the checker slice? Could be as simple as adding two "good" checks and doing a request to see that they are both working before adding the "bad" check.

Reply:
Good points, added more tests.